nixos/modules/services/hardware/display.nix

```diff
@@ -0,0 +1,165 @@
+        };
+    };
+
+    hardware.display.edid.linuxhw = lib.mkOption {
+      type = with lib.types; attrsOf (listOf str);
+      default = { };
+      description = lib.mdDoc ''
+        Exposes EDID files from https://github.com/linuxhw/EDID
+
+        Attribute names will be mapped to EDID filenames `<NAME>.bin`.
+
+        Attribute values are lists of `awk` regexp patterns that (together) must match
+        exactly one entry in either AnalogDisplay.md or DigitalDisplay.md files.
+      '';
+      example = lib.literalExpression ''
+        {
+          PG278Q_2014 = [ "PG278Q" "2014" ];
+        }
+      '';
+      apply = displays:
+        if displays == { } then null else
+        pkgs.linuxhw-edid-fetcher.override { inherit displays; };
+    };
+
+    hardware.display.edid.modelines = lib.mkOption {
+      type = with lib.types; attrsOf str;
+      default = { };
+      description = lib.mdDoc ''
+        Attribute set of XFree86 Modelines automatically converted
+        and exposed as `edid/<name>.bin` files in initrd.
+        See for more information:
+        - https://en.wikipedia.org/wiki/XFree86_Modeline
+      '';
+      example = lib.literalExpression ''
+        {
+          "PG278Q_60" = "    241.50   2560 2608 2640 2720   1440 1443 1448 1481   -hsync +vsync";
+          "PG278Q_120" = "   497.75   2560 2608 2640 2720   1440 1443 1448 1525   +hsync -vsync";
+          "U2711_60" = "     241.50   2560 2600 2632 2720   1440 1443 1448 1481   -hsync +vsync";
+        }
+      '';
+      apply = modelines:
+        if modelines == { } then null else
+        pkgs.edid-generator.overrideAttrs {
+          clean = true;
+          modelines = lib.trivial.pipe modelines [
+            (lib.mapAttrsToList (name: value:
+              lib.throwIfNot (builtins.stringLength name <= 12) "Modeline name must be 12 characters or less"
+                ''Modeline "${name}" ${value}''
+            ))
+            (builtins.map (line: "${line}\n"))
+            (lib.strings.concatStringsSep "")
+          ];
+        };
+    };
+
+    hardware.display.outputs = lib.mkOption {
+      type = lib.types.attrsOf (lib.types.submodule ({
+        options = {
+          edid = lib.mkOption {
+            type = with lib.types; nullOr str;
+            default = null;
+            description = lib.mdDoc ''
+              An EDID filename to be used for configured display, as in `edid/<filename>`.
+              See for more information:
+              - `hardware.display.edid.packages`
+              - https://wiki.archlinux.org/title/Kernel_mode_setting#Forcing_modes_and_EDID
+            '';
+          };
+          mode = lib.mkOption {
+            type = with lib.types; nullOr str;
+            default = null;
+            description = lib.mdDoc ''
+              An `video` kernel parameter (framebuffer mode) configuration for the specific output:
+
+                 <xres>x<yres>[M][R][-<bpp>][@<refresh>][i][m][eDd]
```

Comment:
[Code blocks should be indented by 4 spaces](https://daringfireball.net/projects/markdown/syntax#precode) relative to the surrounding text.